packages/hawtio/src/plugins/camel/routes/CamelRoutes.tsx

```diff
@@ -0,0 +1,313 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import { CamelRoute, routesService } from '@hawtiosrc/plugins/camel/routes-service'
+import { Caption, TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
+import {
+  Button,
+  Card,
+  CardBody,
+  Dropdown,
+  DropdownItem,
+  KebabToggle,
+  Modal,
+  ModalVariant,
+  Page,
+  Text,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+} from '@patternfly/react-core'
+import { AsleepIcon, InfoCircleIcon, PlayIcon, Remove2Icon } from '@patternfly/react-icons'
+import { eventService } from '@hawtiosrc/core'
+import { workspace } from '@hawtiosrc/plugins/shared'
+
+export const CamelRoutes: React.FunctionComponent = () => {
+  const { selectedNode } = useContext(CamelContext)
+  const [routes, setRoutes] = useState<CamelRoute[]>([])
+  const [selected, setSelected] = useState<Map<string, boolean>>(new Map())
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isConfirmDeleteOpen, setIsConfirmDeleteOpen] = useState(false)
+  const getRouteAttributes = () => {
+    routesService.getRoutesAttributes(selectedNode).then(routes => {
+      const tmpSelected = new Map(selected)
+      routes.forEach(route => {
+        selected.set(route.RouteId, selected.get(route.RouteId) ?? false)
+      })
+      setSelected(tmpSelected)
+      setRoutes(routes)
+    })
+  }
+  useEffect(() => {
+    getRouteAttributes()
+  }, [])
```

Comment:
`getRoutesAttributes()` should probably be placed inside the `useEffect` since the linter will complain that it is not included as a dependency. By simply adding it to the dependencies array, an infinite loop will result and it will be necessary to surround the function in a `[useCallback](https://react.dev/reference/react/useCallback)`. However, this has ended up being a lot of extra work when the entire function can be moved into the `useEffect`.

The dependency array should contain the `selectedNode`.